### DISCUSSION TITLE
Prevent dwarves from entering area?

### DISCUSSION THREAD

**Discussion by dethstryke01** [DFHack/dfhack#4607] [Q&A]
I need a way to draw a rectangle over the map to completely prevent dwarves from entering that area. Is there some way to do this? I've read the DFhack tools list and I don't see this thing. I'm prepared to quit playing Dwarf Fortress in frustration because in my current game there is a bug that causes my veteran military dwarves to enter my trap corridor during sieges to pick up equipment that does not exist, causing them to die. I love this game, but this specific bug is currently making the game unplayable for me. I have tried forbidding/autodumping all possible items, all imaginable shenanigans involving burrows, etc. The dwarves walk into my trap corridor on a "pickup equipment" task to a tile where there is no equipment despite the fact that they have been confined within a burrow, and then they just stand there refusing "station" orders until they see enemies entering the trap corridor (and then it's all over for them). I've seen this problem multiple times in multiple different fortresses, and I need this to stop happening. Any help would be greatly appreciated. 

Here is a video documenting this bug. In the beginning I show that both civilians and members of the military are properly confined within burrows and that all members of the military are properly equipped with everything except backpacks. At 3:00, two dwarves then follow pickup equipment orders to empty tiles in the middle of my trap corridor, and it then becomes impossible to give them orders to run away from the invaders. This bug has now been reported to the official DF bug reporting system, I just need some way to stop this problem from happening again. 

https://youtu.be/u2pxktMEbYE?t=181

**Comment by myk002** [comment]
I'm not sure what's going on there with the soldier trying to pick up an item in an empty hallway. If you open the dwarf in `gui/gm-editor`, can you get any details on the item he thinks he's trying to pick up? The item should be attached to the dwarf's current job.

"Confining to burrow" is only a thing when a civalert is active, and you don't need to assign anyone to a burrow for that. Otherwise, burrows only restrict where dwarves can take jobs, not where they can walk.

There is also `gui/teleport` (hotkey: Ctrl-Shift-T) that can teleport your dwarf away from the hallway, but I understand that it would be much preferable to not have this situation happen in the first place.
